packages/dd-trace/src/appsec/iast/analyzers/sql-injection-analyzer.js

```diff
@@ -78,12 +77,7 @@ class SqlInjectionAnalyzer extends InjectionAnalyzer {
 
   _report (value, context, dialect) {
```

Comment:
`sql-injection-analyzer` is overriding `analyze`,  `_reportIfVulnerable`, `_report`, etc methods in order to pass the dialect.
I've checked that no other analyzer overrides them but could we add pass a `metadata` o `extra` argument to these methods in `vulnerability-analyzer` to share the 'extra' info to avoid overriding?

Reply:
I made more changes to overwrite less methods as you suggested